script.js

```diff
@@ -98,19 +134,175 @@ var getDefaultWinLossMessage = function () {
+      (playerObject == REVERSED_SCISSORS && computerObject == SCISSORS) ||
+      (playerObject == REVERSED_PAPER && computerObject == PAPER) ||
+      (playerObject == REVERSED_STONE && computerObject == STONE)
+    );
+  }
+  // determine draw conditions for regular sps. Return a boolean
+  return playerObject == computerObject;
+};
+
+/**
+ * Check whether player's inputs are valid
+ * @param {string} userInput
+ */
+var validateObjectInput = function (userInput) {
+  //if user is playing reversed SPS, validate  input accordingly
+  if (mode == REVERSED_SPS) {
+    return (
+      userInput == REVERSED_SCISSORS ||
+      userInput == REVERSED_PAPER ||
+      userInput == REVERSED_STONE
+    );
+  }
+  // if use is playing regular SPS, validate input accordingly
+  return userInput == SCISSORS || userInput == PAPER || userInput == STONE;
+};
+
+/**
+ * Check whether player's input on the mode is valid
+ */
+var validateModeInput = function (userInput) {
   return (
-    playerObject == computerObject ||
-    (playerObject == REVERSED_SCISSORS && computerObject == SCISSORS) ||
-    (playerObject == REVERSED_PAPER && computerObject == PAPER) ||
-    (playerObject == REVERSED_STONE && computerObject == STONE)
+    userInput == REGULAR ||
+    userInput == REVERSED_SPS ||
+    userInput == KOREAN_SPS ||
+    userInput == COM_V_COM
   );
 };
 
 /**
- * Play SPS with user input, return game result.
+ * Create a string message saying that the user's  input is invalid
+ */
+var createResponseForInvalidInput = function () {
+  // if the mode is 'reversed sps', return the relevant instructions
+  if (mode == REVERSED_SPS) {
+    return (
+      'Please input 1 of ' +
+      REVERSED_SCISSORS +
+      ', ' +
+      REVERSED_PAPER +
+      ', or ' +
+      REVERSED_STONE +
+      ' to play the game.'
+    );
+  }
+  // return the instructions for regular sps
+  return (
+    'Please input 1 of ' +
+    SCISSORS +
+    ', ' +
+    PAPER +
+    ', or ' +
+    STONE +
+    ' to play the game.'
+  );
+};
+
+/**
+ *  Set a fn that returns the instructions for the relevant game
+ */
+var createPlayingInstructions = function () {
+  var preamble =
+    'Thank you for playing, ' + userName + '! <br> To begin, enter ';
+
+  if (mode == COM_V_COM) {
+    preamble =
+      'Thank you for playing, ' +
+      userName +
+      '! <br> To begin, click submit to roll scissors, paper or stone ';
+  }
+
+  // if it is the reversed mode, prompt user to enter 'reversed scissors', 'reversed paper', ...
+  if (mode == REVERSED_SPS) {
+    return (
+      preamble +
+      REVERSED_SCISSORS +
+      ', ' +
+      REVERSED_PAPER +
+      ', or ' +
+      REVERSED_STONE +
+      '.'
+    );
+  }
+  // if it is not the reversed mode, propmpt user to enter 'scissors', 'paper', or 'stone'
+  return preamble + SCISSORS + ', ' + PAPER + ', or ' + STONE + '.';
+};
+
+/*
+ * Set a function that crafts an appropriate output message to display when each round of the game ends
+ */
```

Comment:
Let's be consistent and use JSDoc function-level comments. For function-level multi-line comments, always use `/** ... */` (with 2 stars). `/* ... */` comments (with 1 star) are for inline multi-line comments.